tests/page/page-request-intercept.spec.ts

```diff
@@ -206,6 +206,24 @@ it('should fulfill intercepted response using alias', async ({ page, server, isE
   expect(response.headers()['content-type']).toContain('text/html');
 });
 
+it('should support timeout option in route.fetch', async ({ page, server, isElectron, isAndroid }) => {
+  it.fixme(isElectron, 'error: Browser context management is not supported.');
+  it.skip(isAndroid, 'The internal Android localhost (10.0.0.2) != the localhost on the host');
+
+  server.setRoute('/slow', (req, res) => {
+    res.writeHead(200, {
+      'content-length': 4096,
+      'content-type': 'text/html',
+    });
+  });
+  await page.route('**/*', async route => {
+    const error = await route.fetch({ timeout: 10 }).catch(e => e);
+    expect(error.message).toContain(`Request timed out after 10ms`);
+  });
+  const error = await page.goto(server.PREFIX + '/slow', { timeout: 20 }).catch(e => e);
```

Comment:
I added a small (but larger than the inner) timeout here so it wouldn't slow down the tests waiting for it to time out, but if there's a better way?

Reply:
This is good. I'd only ask to bump 10 to 1000 and 20 to 2000. Given our experience on GHA bots, 10/20 is waaay too small for non-flaky tests.